tests/EncryptedFS.concurrent.test.ts

```diff
@@ -240,120 +293,135 @@ describe(`${EncryptedFS.name} Concurrency`, () => {
   });
   describe('concurrent file writes', () => {
     test('EncryptedFS.write on multiple file descriptors', async () => {
-      // Concurrent writes of the same length results in "last write wins"
-      let fds: Array<FdIndex> = [
-        await efs.open('test', constants.O_RDWR | constants.O_CREAT),
-        await efs.open('test', constants.O_RDWR | constants.O_CREAT),
-      ];
-      let contents = ['one', 'two'];
-      let promises: Array<Promise<any>>;
-      promises = [];
-      for (let i = 0; i < 2; i++) {
-        promises.push(efs.write(fds[i], contents[i]));
-      }
-      await Promise.all(promises);
-      expect(['one', 'two']).toContainEqual(
-        await efs.readFile('test', { encoding: 'utf-8' }),
-      );
-      for (const fd of fds) {
-        await efs.close(fd);
-      }
-      // Concurrent writes of different length results in "last write wins" or a merge
-      fds = [
-        await efs.open('test', constants.O_RDWR | constants.O_CREAT),
-        await efs.open('test', constants.O_RDWR | constants.O_CREAT),
-      ];
-      contents = ['one1', 'two'];
-      promises = [];
-      for (let i = 0; i < 2; i++) {
-        promises.push(efs.write(fds[i], contents[i]));
-      }
-      expect(['one1', 'two', 'two1']).toContainEqual(
-        await efs.readFile('test', { encoding: 'utf-8' }),
+      const contents = ['one', 'two'];
+      await fc.assert(
+        fc.asyncProperty(fc.scheduler(), async (s) => {
+          const gate = gateFactory(s);
+          const fds: Array<FdIndex> = [
+            await efs.open('test', constants.O_RDWR | constants.O_CREAT),
+            await efs.open('test', constants.O_RDWR | constants.O_CREAT),
+          ];
+
+          const prom = Promise.all([
+            gate(() => efs.write(fds[0], contents[0])),
+            gate(() => efs.write(fds[1], contents[1])),
+          ]);
+          await s.waitAll();
+          await prom;
+
+          expect(
+            await efs.readFile('test', { encoding: 'utf-8' }),
+          ).toHaveLength(3);
```

Comment:
Change this to a regular expression. Like` expect().toMatch(/one|two/)`.

Reply:
Use regular expressions for any kind of non-deterministic results.

Reply:
I've added the check back in.